test/lib/ansible_test/_internal/sanity/validate_modules.py

```diff
@@ -48,6 +50,39 @@
 )
 
 
+# from ansible.constants import DOCUMENTABLE_PLUGINS
+DOCUMENTABLE_PLUGINS = (
```

Comment:
Could this opt-in (rather than opt-out) mean we don't validate documentation if there is a new plugin type on the future?

Reply:
Yes. I originally wanted to import `DOCUMENTABLE_PLUGINS` from `ansible.constants`, but that seems to be somewhere between hard to impossible. I tried something similar to https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_internal/util.py#L218-L234 but unfortunately `constants` depends on a large chunk of Ansible. @mattclay do you have an idea how to solve this in a good way?

Reply:
We have similar lists in other places as well:

https://github.com/ansible/ansible/blob/fc08c1f3c554c9cd4ad39f846d123408de19ad2a/docs/docsite/Makefile#L35

https://github.com/ansible/ansible/blob/fc08c1f3c554c9cd4ad39f846d123408de19ad2a/test/lib/ansible_test/_internal/sanity/ansible_doc.py#L52-L61

There are other locations that also list plugins types, mostly for non-documentation purposes. I don't think we'll be unifying all the various lists anytime soon.

@felixfontein What's the issue with just importing `ansible.constants` directly from within validate-modules itself?

Reply:
@mattclay then validate-modules itself needs to figure out which path belongs to which plugin class. That's doable, but it is a lot easier to do in ansible-test directly since it already knows the paths (`data_context().content.plugin_paths`).

Reply:
@felixfontein We could have ansible-test pass the plugin list and paths to validate-modules when invoking it.

Reply:
@mattclay true, but I think it would make validate-modules more complicated (also more complicated to use stand-alone, though I don't know if anyone actually does that).

What do you prefer?

Reply:
I yesterday learned that the ansible-doc sanity test has a similar hard-coded list of plugin types that are not documentable. I've moved that list to a common constant also use it in here now, so we have only one place left where (un-)documentable plugins are listed.

Reply:
(See 75c4ffb0803e9f20268ca7cd975bce8922d19548)